gbpservice/nfp/proxy_agent/proxy/proxy.py

```diff
@@ -184,7 +186,9 @@ def recv(self):
         return None
 
     def send(self, data):
-        self._socket.send(data)
+        self._socket.setblocking(1)
+        self._socket.sendall(data)
```

Comment:
Is this logic because we want to make send blocking but recv non-blocking? If so, why?